lib/tasks/migration.rake

```diff
@@ -216,10 +216,15 @@ namespace :migration do
     @ingest_batch = Batch.find_or_create(@ingest_batch_id)
     MigrationLogger.info "Ingest Batch ID #{@ingest_batch_id}"
     #for each metadata file in the migration directory
-    Dir.glob(metadata_dir+"/uuid_*.xml") do |file|
+    allfiles = Dir.glob(metadata_dir+"/uuid_*.xml")
+    filecount = allfiles.select { |file| File.file?(file) }.count
+    MigrationLogger.info "Files to process: " + filecount.to_s
+    thisfile = 1
+    allfiles.sort.each do |file|
```

Comment:
not important, but a neat trick -- instead of defining a counter outside the loop, and manually incrementing it each iteration, you can do:

```ruby
allfiles.sort.each_with_index do |file, index|
...
end
```

The only caveat is that the indexing is zero-based.